diff --git a/run_alphafold.py b/run_alphafold.py
index 4ddb313..21ac4bd 100644
--- a/run_alphafold.py
+++ b/run_alphafold.py
@@ -98,6 +98,8 @@ flags.DEFINE_enum('model_preset', 'monomer',
                   'Choose preset model configuration - the monomer model, '
                   'the monomer model with extra ensembling, monomer model with '
                   'pTM head, or multimer model')
+flags.DEFINE_boolean('prepare', False, 'Run prepare feature dictionary step only '
+                     '(does not require GPU)')
 flags.DEFINE_boolean('benchmark', False, 'Run multiple JAX model evaluations '
                      'to obtain a timing that excludes the compilation time, '
                      'which should be more indicative of the time required for '
@@ -179,6 +181,9 @@ def predict_structure(
   with open(features_output_path, 'wb') as f:
     pickle.dump(feature_dict, f, protocol=4)
 
+  if FLAGS.prepare:
+    return
+
   unrelaxed_pdbs = {}
   relaxed_pdbs = {}
   relax_metrics = {}
